<?xml version="1.0"?>
<launch>
  <arg name="device" default="R2300"/>

  <include file="$(find pf_description)/launch/pf_bringup.launch" >
    <arg name="scanner" value="r2300" />
  </include>

  <!-- R2300 Driver -->
  <node pkg="pf_driver" type="ros_main" name="r2300_node" output="screen">
    <param name="device" value="$(arg device)"/>
    <rosparam file="$(find pf_driver)/config/r2300_params.yaml" command="load"/>
  </node>

  <node pkg="tf" type="static_transform_publisher" name="ring1" args="0 0 0 0 0.0785 0 base_link scanner_1 10" />
	<node pkg="tf" type="static_transform_publisher" name="ring2" args="0 0 0 0 0.0262 0 base_link scanner_2 10" />
	<node pkg="tf" type="static_transform_publisher" name="ring3" args="0 0 0 0 -0.0785 0 base_link scanner_3 10" />
	<node pkg="tf" type="static_transform_publisher" name="ring4" args="0 0 0 0 -0.0262 0 base_link scanner_4 10" />
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find pf_driver)/rviz/r2300.rviz" />
</launch>